src/HotChocolate/AspNetCore/src/AspNetCore/HttpGetMiddleware.cs

```diff
@@ -88,10 +88,28 @@ private async Task HandleRequestAsync(HttpContext context)
                 result = QueryResultBuilder.CreateError(error);
             }
 
-            // in any case we will have a valid GraphQL result at this point that can be written
-            // to the HTTP response stream.
-            Debug.Assert(result is not null!, "No GraphQL result was created.");
-            await WriteResultAsync(context.Response, result, statusCode, context.RequestAborted);
+            try
+            {
+                // in any case we will have a valid GraphQL result at this point that can be written
+                // to the HTTP response stream.
+                Debug.Assert(result is not null!, "No GraphQL result was created.");
+                await WriteResultAsync(context.Response, result, statusCode,
```

Comment:
@PascalSenn we did all the work on memory pooling.... but we never returned the results. Just for subscriptions the results were correctly disposed.